src/Kiota.Builder/Validation/OpenApiSchemaComparer.cs

```diff
@@ -12,39 +12,71 @@ internal class OpenApiSchemaComparer : IEqualityComparer<OpenApiSchema>
 {
     private static readonly OpenApiDiscriminatorComparer discriminatorComparer = new();
     private static readonly OpenApiAnyComparer openApiAnyComparer = new();
+    private static readonly KeyValueComparer<string, OpenApiSchema> schemaMapComparer = new(StringComparer.Ordinal, new OpenApiSchemaComparer());
     /// <inheritdoc/>
     public bool Equals(OpenApiSchema? x, OpenApiSchema? y)
     {
-        return x == null && y == null || x != null && y != null && GetHashCode(x) == GetHashCode(y);
+        return EqualsInternal(x, y);
     }
     /// <inheritdoc/>
     public int GetHashCode([DisallowNull] OpenApiSchema obj)
     {
-        return GetHashCodeInternal(obj, new());
+        var hash = new HashCode();
+        GetHashCodeInternal(obj, new(), ref hash);
+        return hash.ToHashCode();
     }
-    private static int GetHashCodeInternal([DisallowNull] OpenApiSchema obj, HashSet<OpenApiSchema> visitedSchemas)
+
+    private bool EqualsInternal(OpenApiSchema? x, OpenApiSchema? y)
+    {
+        if (x is null || y is null) return object.Equals(x, y);
+        if (x.Deprecated != y.Deprecated) return false;
```

Comment:
should this be simplified to a single conditional expression to avoid the additional ifs and returns?